tests/spec_decode/test_multi_step_worker.py

```diff
@@ -173,7 +173,6 @@ def test_same_output_for_multi_step():
         block_size,
         num_gpu_blocks,
         seed,
-        model_runner_cls=TP1DraftModelRunner,
```

Comment:
Wait I just found this. Why we remove this line? I suppose this test is for TP1 draft model runner?

Reply:
Actually the current main brach fails this test as well. 
This is because
(1) the test did not configure MultiStepWorker correctly. It did not explicitly persist GPU tensors. When used with SpecDecodeWorker, it configures this sampler parameter as shown [here](https://github.com/vllm-project/vllm/blob/1cabfcefb64a489c8ff9dcb289b4dd47cf8f89cf/vllm/spec_decode/spec_decode_worker.py#L306).
(2) not sure which PR modifies the sampler output, the current test does not output correct CPU sampler output. Therefore, even if (1) is fixed, this test will fail when extracting the final output because it cannot get the output tokens for comparison.

We need to either change this test, or maybe just use the default model runner to pass this test.

Reply:
reverse this change for now, will fix it in future prs.